swap/swap.go

```diff
@@ -59,15 +63,38 @@ func (s *Swap) Add(amount int64, peer *protocols.Peer) (err error) {
 	if err != nil && err != state.ErrNotFound {
 		return
 	}
+
+	//check if balance with peer is over the disconnect threshold
+	if s.balances[peer.ID()] >= s.disconnectThreshold {
```

Comment:
Here again the threshold crossing is only in positive sense. 

But if the local node is having a negative balance with the remote peer, then this threshold would be crossed with a negative value. The remote node will drop the local node, but this node may need to also take some action - at least **log** this I think.

Reply:
Noted, I'll add a log entry for the negative case.

Should I also add logs in the positive cases for crossing the thresholds? Or is issuing the cheque enough since it will have its own logs?

Reply:
Good question. I think it would be good to have a log entry as well here.
What do you think?

I am even thinking of a completely independent log level for Swap logs...like audit logs....
e.g. `log.Audit(text)`

Reply:
+1 for the audit logs.

I wouldn't add a positive log if the cheque request call already includes this. But that's on your side for now :-)